src/features/commentsSlice.ts

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable no-param-reassign */
+import {
+  createSlice,
+  PayloadAction,
+  createAsyncThunk,
+} from '@reduxjs/toolkit';
+import { getPostComments } from '../api/comments';
+import { Comment } from '../types/Comment';
+
+type CommentsState = {
+  items: Comment[],
+  loaded: boolean,
+  hasError: boolean,
+};
+
+const initialState: CommentsState = {
+  items: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const init = createAsyncThunk(
+  'comments/fetch',
+  (postId: number) => getPostComments(postId),
+);
+
+const commentsSlice = createSlice({
+  name: 'comments',
+  initialState,
+  reducers: {
+    add: (state, action: PayloadAction<Comment>) => {
+      state.items.push(action.payload);
+    },
+    remove: (state, action: PayloadAction<number>) => {
+      state.items = state.items.filter(item => item.id !== action.payload);
+    },
+    setError: state => {
+      state.hasError = true;
+    },
+  },
+  extraReducers: builder => {
+    builder.addCase(init.pending, state => {
+      state.loaded = true;
+    });
+
+    builder.addCase(
+      init.fulfilled,
+      (state, action: PayloadAction<Comment[]>) => {
+        state.items = action.payload;
+        state.loaded = false;
+      },
+    );
```

Comment:
Better to write in such way (without `builder.addCase` duplicate, simple: `addCase(...).addCase(...)`):
```
const counterReducer = createReducer(initialState, (builder) => {
  builder
    .addCase(increment, (state, action) => {
      state.value++
    })
    .addCase(decrement, (state, action) => {
      state.value--
    })
    .addCase(incrementByAmount, (state, action) => {
      state.value += action.payload
    })
})
```